documentation/project-docs/breaking-change-guidelines.md

```diff
@@ -0,0 +1,81 @@
+# SDK Breaking Change and Diagnostic Guidelines
+
+This document provides guidelines for introducing new diagnostics or breaking changes to the .NET SDK, which configuration knobs are available, and what criteria should guide the decision around severity, timeline for introduction, and deprecation of diagnostics.
+
+## Overall procedure
+
+In general, we want to make updating the .NET SDK as smooth as possible for developers. This means:
+
+* introducing new changes in a staged/gradual way
+* trying to tie opinionated analyzers/diagnostics to a mechanism that requires explicit user opt-in
+* providing a way to opt out of a change entirely
+
+## Kinds of changes
+
+There are many kinds of breaking changes that can ship in the .NET SDK, such as:
+
+* New MSBuild warnings and errors (props/targets)
+* New NuGet warnings and errors. 
+  * For example, NuGet Audit.
+* Roslyn Analyzers and CodeFixes
+  * This includes trimming/ILLink analyzers and codefixes
+* Behavioral/implementation changes
+  * MSBuild engine changes like MSBuild Server
+  * Implementation changes for MSBuild Tasks
+  * NuGet Restore algorithm enhancements
+  * Changes to DotNet CLI grammar
+  * Changes to defaults in CLI flags that impact behavior
+    * For example, `--configuration` flag defaulting to `Release` instead of `Debug`
+
+## Configuration Knobs
+
+The following knobs are available to enable/disable these changes (some may not apply to all kinds of changes):
```

Comment:
Change waves and LangVer also apply to the tooling ecosystem.

Reply:
LangVer - I read https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version, and it looks like it's the same as TFM for .NET Core. So, I'm wondering in which cases langver would be used instead of TFM.

Change waves - I found a description here: https://learn.microsoft.com/en-us/visualstudio/msbuild/change-waves?view=vs-2022. Moving forward, do we recommend people using change waves or using the new SdkAnalysisLevel property?

@baronfel

Reply:
LangVersion _is_ often the same as the numeric part of TFM, but that's purely a convenience. Users often set it explicitly when
* multitargeting and wanting to ensure a consistent level of analysis across TFMs
* when they want to use `preview` language features

Reply:
Done.